cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -0,0 +1,869 @@
+                                p.complete(freshClusterRebalance.getStatus());
+                            }
+                        } else {
+                            log.debug("{}: Rebalance resource was deleted, stopping the request time", reconciliation);
+                            vertx.cancelTimer(t);
+                            p.complete();
+                        }
+                    } else {
+                        log.error("{}: Cruise Control getting rebalance resource failed", reconciliation, getResult.cause());
+                        vertx.cancelTimer(t);
+                        p.fail(getResult.cause());
+                    }
+                });
+
+            });
+        } else {
+            p.complete(clusterRebalance.getStatus());
+        }
+        return p.future();
+    }
+
+    /**
+     * This method handles the transition from {@code Stopped} state.
+     * If the user set strimzi.io/rebalance=refresh annotation, it calls the Cruise Control API for requesting a new rebalance proposal.
+     * If the proposal is immediately ready, the next state is {@code ProposalReady}.
+     * If the proposal is not ready yet and Cruise Control is still taking care of processing it, the next state is {@code PendingProposal}.
+     * If the user sets any other values for the strimzi.io/rebalance, it is just ignored.
+     *
+     * @param reconciliation Reconciliation information
+     * @param host Cruise Control service to which sending the rebalance proposal request
+     * @param apiClient Cruise Control REST API client instance
+     * @param rebalanceAnnotation The current value for the strimzi.io/rebalance annotation
+     * @param rebalanceOptionsBuilder builder for the Cruise Control REST API client options
+     * @return a Future with the next {@code KafkaRebalanceStatus} bringing the state
+     */
+    private Future<KafkaRebalanceStatus> onStop(Reconciliation reconciliation,
+                                                String host, CruiseControlApi apiClient,
+                                                RebalanceAnnotation rebalanceAnnotation,
+                                                RebalanceOptions.RebalanceOptionsBuilder rebalanceOptionsBuilder) {
+        if (rebalanceAnnotation == RebalanceAnnotation.refresh) {
+            return requestRebalance(reconciliation, host, apiClient, true, rebalanceOptionsBuilder);
+        } else {
+            log.warn("{}: Ignore annotation {}={}", reconciliation, ANNO_STRIMZI_IO_REBALANCE, rebalanceAnnotation);
+            return Future.succeededFuture(new KafkaRebalanceStatusBuilder()
+                    .addNewCondition().withNewType(State.Stopped.toString()).endCondition().build());
+        }
+    }
+
+    private Future<Void> reconcileRebalance(Reconciliation reconciliation, KafkaRebalance kafkaRebalance) {
+        if (kafkaRebalance == null) {
+            log.info("{}: Rebalance resource deleted", reconciliation);
```

Comment:
So ... shouldn't thi have any impact on things? Such as stopping the rebalance or proposal preparation?

Reply:
This code is mostly related to the trigger periodic reconcile. When a resource is deleted in the middle of a proposal computation or actual rebalancing, the resource deletion is handled there stopping polling mechanism and ongoing task. See `onPendingProposal` and `onRebalancing`.